src/stomach.cpp

```diff
@@ -237,138 +125,90 @@ void stomach_contents::ingest( player &p, item &food, int charges = 1 )
     mod_calories( p.kcal_for( comest ) );
 }
 
-void stomach_contents::absorb_water( player &p, units::volume amount )
+void stomach_contents::ingest( nutrients ingested )
 {
-    if( water < amount ) {
-        amount = water;
-        water = 0_ml;
-    } else {
-        water -= amount;
-    }
-    if( p.get_thirst() < -100 ) {
-        return;
-    } else if( p.get_thirst() - units::to_milliliter( amount / 5 ) < -100 ) {
-        p.set_thirst( -100 );
+    contents += ingested.solids;
+    water += ingested.water;
+    calories += ingested.kcal;
+    for( auto &vit : ingested.vitamins ) {
+        vitamins[vit.first] += vit.second;
     }
-    p.mod_thirst( -units::to_milliliter( amount ) / 5 );
 }
 
-void stomach_contents::absorb_kcal( int amount )
+nutrients stomach_contents::digest( player &owner, int five_mins, int half_hours )
```

Comment:
is there a reason `player &` isn't `const` here?

Reply:
Going further, I replaced `player &` here with `const needs_rates &` (which I think is more similar to how you were handling it before, not sure why I changed it).